test_util/aws.py

```diff
@@ -170,30 +208,28 @@ def empty_and_delete_s3_bucket_from_stack(self):
         bucket.delete()
         log.info('Delete successfully triggered for {}'.format(self.stack.stack_name))
 
-    def get_master_ips(self, state_list=None):
-        instances = self.get_group_instances(['MasterServerGroup'], state_list)
+    def get_master_ips(self):
+        instances = self.get_group_instances('MasterServerGroup')
         return instances_to_hosts(instances)
 
-    def get_public_agent_ips(self, state_list=None):
-        instances = self.get_group_instances(['PublicSlaveServerGroup'], state_list)
+    def get_public_agent_ips(self):
+        instances = self.get_group_instances('PublicSlaveServerGroup')
         return instances_to_hosts(instances)
 
-    def get_private_agent_ips(self, state_list=None):
-        instances = self.get_group_instances(['SlaveServerGroup'], state_list)
+    def get_private_agent_ips(self):
+        instances = self.get_group_instances('SlaveServerGroup')
         return instances_to_hosts(instances)
 
-    @retry_boto_rate_limits
-    def get_group_instances(self, group_list, state_list=None):
+    def get_group_instances(self, group_name):
         """Returns a list of dictionaries that describe currently running instances of group
+        going through all ASGs is
         """
-        filters = [{'Name': 'tag:aws:cloudformation:stack-name', 'Values': [self.stack.stack_name]}]
-        if group_list:
-            filters.append({'Name': 'tag:aws:cloudformation:logical-id', 'Values': group_list})
-        if state_list:
-            filters.append({'Name': 'instance-state-name', 'Values': state_list})
-        reservations = self.boto_wrapper.client('ec2').describe_instances(
-            Filters=filters)['Reservations']
-        return [instance for res in reservations for instance in res['Instances']]
+        instance_ids = []
+        for asg in self.get_auto_scaling_groups():
```

Comment:
any reason not to use: `asg_resource = self.stack.Resource(group_name)` to get the CF Resource, then `self.get_autoscaling_instances(self, asg_resource.physical_resource_id)`?

which should in theory return a list of https://boto3.readthedocs.io/en/latest/reference/services/ec2.html#instance